posts/2022/12-17-first-dev-job.md

```diff
@@ -0,0 +1,105 @@
+
+My main Git tips are:
+
+1. Rebase your branches off main as soon as something new gets merged into the main branch. Nobody wants to deal with a merge conflict on a branch that is multiple commits behind the main branch.
+
+2. Try not to have multiple branches/PRs open. Finish one thing, put it in for review and move on to the next. Unless of course, you are stuck or a PR with higher priority than your current one has been assigned to you. 
+
+3. Don't copy code between branches. Just don't.
+
+
+# 3. A good process
+
+
+One thing I learned from my dev husband (Hi [Alan](https://alanvardy.com/)!) is the importance of having a good process in place. By that, I mean finding a good way to tackle work and responsibilities consistently every day. A routine if you so will. Everybody is different, so what works for me might not be the best fit for you, but here are some more general tips:
+
+1.  Find your prime work times. I'm that weirdo who likes early mornings (and also Mondays), so I try to get some deep work in early in the morning before my family wakes up. I've noticed that my brain stops functioning somewhere around 5 PM, which is why I try to stop before it grinds to a complete standstill. 
+
+
+2. Establish routines. What do you do when you first switch on the computer? What works best for you? Find out and do it consistently. I have made it a habit to check the main branch of the repo before I do any work and rebase my branches if need be. I also start work on a new ticket in the same way every time. Having these routines in place is one less thing to think about.
+
+3. Make notes for yourself. I find it helpful to write down the next steps for the ticket I am working on and to leave notes for my morning self every evening before I shut down for the night. 
+
+
+# 4. Set-up
+
+
+Less than two months into the new job, and I don't recognize myself anymore. I vividly remember telling somebody less than half a year ago that mechanical keyboards were weird and that I hated the sound of my husband typing (Cherry Brown). Guess who is clicking away on Cherry Blues now? 
+I also invested in a bigger screen. But you know what: it's worth it. Having the ability to see my ticket, my code editor (in a larger window now) and what I'm producing on localhost has improved my life tremendously.
+Here is a picture of my current setup. Luckily my little desk has a drawer. Otherwise, I would have needed a bigger desk. Shout-out to my son, who let me borrow his Squishmallow as a rubber ducky substitute. Now I am slightly less crazy because I'm talking to the Squishmallow and not myself when I'm coding.
```

Comment:
```suggestion
Here is a picture of my current setup. Luckily my little desk has a drawer. Otherwise, I would have needed a bigger desk. Shout out to my son, who let me borrow his Squishmallow as a rubber ducky substitute. Now I am slightly less crazy because I'm talking to the Squishmallow and not myself when I'm coding.
```